src/app/components/AddEvaluationForm/AddEvaluationForm.tsx

```diff
@@ -0,0 +1,171 @@
+import styled from 'styled-components';
+import useEvaluations from '../../hooks/useEvaluations';
+import type { Pupil } from '../../types/types';
+import Button from '../Button/Button';
+import Navigation from '../Navigation/Navigation';
+
+type AddEvaluationFormProps = {
+  pupil: Pupil;
+  onSubmit: (category: string, evaluation: string) => void;
+  onCancel: () => void;
+  missingInput: boolean;
+};
+
+export default function AddEvaluationForm({
+  pupil,
+  onSubmit,
+  onCancel,
+}: AddEvaluationFormProps): JSX.Element {
+  const { evaluations } = useEvaluations();
+  const [selectedCategory, setSelectedCategory] = useState('');
+  const [selectedEvaluation, setSelectedEvaluation] = useState('');
+  const [rating, setRating] = useState(0);
+  const [inputError, setInputError] = useState(false);
+  console.log(evaluations);
+
+  function handleRating(rate: number) {
+    setRating(rate);
+  }
+
+  function handleCategory(category: string) {
+    setSelectedCategory(category);
+  }
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    if (selectedCategory === '' || rating === 0 || selectedEvaluation === '') {
+      setInputError(true);
+      return;
+    } else {
+      const category = selectedCategory;
+      const evaluation = selectedEvaluation;
+      onSubmit(category, evaluation);
+      setInputError(false);
+      setSelectedCategory('');
+      setSelectedEvaluation('');
+    }
+  }
+
+  function setName(description: string) {
+    return description.replace('X', pupil.name.first);
+  }
+
+  return (
+    <FormWrapper>
+      <FormContainer onSubmit={handleSubmit}>
+        <label htmlFor="category">Kategorie wählen:</label>
+        {evaluations.map((evaluation, key) => (
+          <CategoryButton
+            key={`${evaluation.name}-${key}`}
+            children={evaluation.name}
+            onClick={() => handleCategory(evaluation.name)}
+            type="button"
+            category={evaluation.name}
+            selectedCategory={selectedCategory}
+          />
+        ))}
+
+        {inputError && selectedCategory === '' && (
+          <SubmitWarning>Bitte geben Sie eine Kategorie ein.</SubmitWarning>
+        )}
+        <label>Bewertung wählen:</label>
+        <Rating
+          onClick={handleRating}
+          ratingValue={rating}
+          size={40}
+          transition
+          fillColor={`var(--color-button)`}
+          emptyColor="gray"
+        />
+
+        {inputError && rating === 0 && (
+          <SubmitWarning>Bitte geben Sie eine Bewertung an.</SubmitWarning>
+        )}
+        <label htmlFor="evaluation">Worturteil:</label>
+        {selectedCategory &&
+          evaluations.map((evaluation) =>
+            evaluation.name === selectedCategory
+              ? evaluation.evaluations.map((evaluation) =>
+                  evaluation.mark === rating
+                    ? evaluation.descriptions.map((description, key) => (
```

Comment:
there is a evaluations array inside evaluation inside evaluations? this is very confusing